mergin/client.py

```diff
@@ -144,6 +145,21 @@ def ignore_file(self, file):
             return True
         return False
 
+    """
+    function to do checkpoint over the geopackage file which was not able to do diff file
+    """
+    def do_sqlite_checkpoint(self, file):
```

Comment:
I would suggest to modify this function a bit:
- pass just file_path string (with absolute path to the .gpkg file)
- move update of the checksum out of this function (so that it only deals with the checkpoint handling)
- move the function to utils.py